src/python/pants/python/python_setup.py

```diff
@@ -204,15 +209,18 @@ def artifact_cache_dir(self):
     def scratch_dir(self):
         return os.path.join(self.get_options().pants_workdir, *self.options_scope.split("."))
 
-    def compatibility_or_constraints(self, compatibility):
-        """Return either the given compatibility, or the interpreter constraints. If interpreter
-        constraints are supplied by the CLI flag, return those only.
+    def compatibility_or_default(self, compatibility):
+        """Return the value `compatibility` if non-None, else this subsystem's interpreter
+        constraints."""
+        return self.get_scalar_value_option("interpreter_constraints", compatibility)
+
+    def compatibility_for_target(self, target):
```

Comment:
Type hints appreciated. I belive this is `Union[Target, TargetAdaptor]` and returns `Tuple[str, ...]`.